File: app/src/main/java/com/wlmtest/forecasttestjava/view/search/CityForecastRecicleViewAdapter.java
package com.wlmtest.forecasttestjava.view.search;

import android.view.LayoutInflater;
import android.view.ViewGroup;
import android.widget.ImageView;
import androidx.annotation.NonNull;
import androidx.databinding.DataBindingUtil;
import androidx.recyclerview.widget.RecyclerView;
import com.squareup.picasso.Picasso;
import com.wlmtest.forecasttestjava.R;
import com.wlmtest.forecasttestjava.databinding.CityForecastItemBinding;
import com.wlmtest.forecasttestjava.viewmodel.MainViewModel;


public class CityForecastRecicleViewAdapter extends RecyclerView.Adapter<CityForecastRecicleViewAdapter.ViewHolder> {

    private MainViewModel mainViewModel;

    public CityForecastRecicleViewAdapter(MainViewModel mainViewModel) {
        this.mainViewModel = mainViewModel;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {

        CityForecastItemBinding cityForecastItemBinding =
                DataBindingUtil.inflate(
                        LayoutInflater.from(parent.getContext()),
                        R.layout.city_forecast_item, parent, false);

        cityForecastItemBinding.setMainViewModel(this.mainViewModel);

        return new ViewHolder(cityForecastItemBinding);

    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {

        holder.cityForecastItemBinding
                .setCityForecastPojo(
                        this.mainViewModel.getCityForecastSearchListMutableLiveData().getValue().get(position));

        try {
            String iconRef = this.mainViewModel.getCityForecastSearchListMutableLiveData().getValue().get(position).getIcon();

            ImageView ivForecastItem = holder.cityForecastItemBinding.ivForecastItem;
            if (iconRef != null && !iconRef.isEmpty()) {
                Picasso.get()
                        .load("http://openweathermap.org/img/w/" + iconRef + ".png")
                        .into(ivForecastItem);
            }
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    @Override
    public int getItemCount() {
        if (this.mainViewModel.getCityForecastSearchListMutableLiveData() == null ||
                this.mainViewModel.getCityForecastSearchListMutableLiveData().getValue() == null) {
            return 0;
        }
        return this.mainViewModel.getCityForecastSearchListMutableLiveData().getValue().size();
    }

    static class ViewHolder extends RecyclerView.ViewHolder {

        public CityForecastItemBinding cityForecastItemBinding;

        public ViewHolder(CityForecastItemBinding cityForecastItemBinding) {

            super(cityForecastItemBinding.getRoot());
            this.cityForecastItemBinding = cityForecastItemBinding;

        }
    }
}

File: app/src/main/java/com/wlmtest/forecasttestjava/view/search/CityForecastRecicleViewAdapter.kt
package com.wlmtest.forecasttestjava.view.search

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import androidx.recyclerview.widget.RecyclerView
import com.squareup.picasso.Picasso
import com.wlmtest.forecasttestjava.R
import com.wlmtest.forecasttestjava.databinding.CityForecastItemBinding
import com.wlmtest.forecasttestjava.viewmodel.MainViewModel


class CityForecastRecicleViewAdapter(private val mainViewModel: MainViewModel) :
    RecyclerView.Adapter<CityForecastRecicleViewAdapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {

        val cityForecastItemBinding = DataBindingUtil.inflate<CityForecastItemBinding>(
            LayoutInflater.from(parent.context),
            R.layout.city_forecast_item, parent, false
        )

        cityForecastItemBinding.mainViewModel = this.mainViewModel

        return ViewHolder(cityForecastItemBinding)

    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {

        holder.cityForecastItemBinding.cityForecastPojo =
            this.mainViewModel.cityForecastSearchList.value!![position]

        try {
            val iconRef = this.mainViewModel.cityForecastSearchList.value!![position].icon

            val ivForecastItem = holder.cityForecastItemBinding.ivForecastItem
            if (iconRef != null && !iconRef.isEmpty()) {
                Picasso.get()
                    .load("http://openweathermap.org/img/w/$iconRef.png")
                    .into(ivForecastItem)
            }
        } catch (e: Exception) {
            e.printStackTrace()
        }

    }

    override fun getItemCount(): Int {
        return if (this.mainViewModel.cityForecastSearchList == null || this.mainViewModel.cityForecastSearchList.value == null) {
            0
        } else this.mainViewModel.cityForecastSearchList.value!!.size
    }

    class ViewHolder(var cityForecastItemBinding: CityForecastItemBinding) :
        RecyclerView.ViewHolder(cityForecastItemBinding.root)
}
